onnxscript/values.py

```diff
@@ -199,6 +199,8 @@ def _usercall(self, *args, **kwargs):
                 new_args.append(tensor.Tensor(a))
             elif isinstance(a, bool):
                 new_args.append(tensor.Tensor(np.array(a)))
+            elif isinstance(a, int):
+                new_args.append(a)
```

Comment:
why not `elif isinstance(a, (bool, int, float)):

Reply:
if use this:
```
 elif isinstance(a, (bool, int, float)):
```

the N in below code will be Tensor(array(3)), so the range(N) will throw exception.

```
for i in range(N):
```

Reply:
According to the code I wrote a while ago at this line: https://github.com/microsoft/onnx-script/blob/main/onnxscript/test/models/signal_dft.py#L364, `range(int(N))` should work if `N` is a tensor. I might be wrong but the change you made is not consistent with the line just above, so I fear another bug arises later.

Reply:
Yes, range(int(N)) works. But from the user perspective, if the 'user' is us, I feel OK. if the 'user' is external user, they may feel weird, because from the context of code, the N is already INT type.
I'm not talking my change is correct or not, I care about user experience, because we have a task from Aaron that we need to give spec about how to let user write script in reasonable way. :)

Reply:
Is it possible to allow a scalar tensor behave like a Python scalar like numpy does?

Reply:
> Is it possible to allow a scalar tensor behave like a Python scalar like numpy does?

Yes, since `range` works for a numpy array, maybe we can do the same. Maybe we just need to add an `__int__` method to our Tensor class? Must be something like that?

Reply:
Looks like it's `__index__` called:

```
>>> for i in range(np.array([1,2])): print("loop", i)
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: only integer scalar arrays can be converted to a scalar index
```

Reply:
#181